src/library/dao/playlistdao.cpp

```diff
@@ -980,18 +982,16 @@ void PlaylistDAO::shuffleTracks(const int playlistId, const QList<int>& position
         trackPositionIds.insert(trackBPosition, trackAId);
         newPositions.swap(newPositions.indexOf(trackAPosition),
                           newPositions.indexOf(trackBPosition));
-        QString swapQuery = "UPDATE PlaylistTracks SET position=%1 "
-                "WHERE position=%2 AND playlist_id=%3";
-        query.exec(swapQuery.arg(QString::number(-1),
-                                 QString::number(trackAPosition),
-                                 QString::number(playlistId)));
-        query.exec(swapQuery.arg(QString::number(trackAPosition),
-                                 QString::number(trackBPosition),
-                                 QString::number(playlistId)));
-        query.exec(swapQuery.arg(QString::number(trackBPosition),
-                                 QString::number(-1),
-                                 QString::number(playlistId)));
 
+        qSwap(playListTrackIds[trackAPosition - 1],
+              playListTrackIds[trackBPosition - 1]);
+    }
+    
+    QString changePositionQuery = "UPDATE PlaylistTracks SET position=%1 "
+        "WHERE id=%2";
+    for(int i = 0; i < playListTrackIds.size(); i++) {
+        query.exec(changePositionQuery.arg(QString::number(i + 1),
+                                           QString::number(playListTrackIds[i])));
```

Comment:
No, this should be right, as any any of the index of the list playListTrackIds store the ID from the table and the index represents the positions, thus the list stores ids at position 1, 2, 3 ..., so it should be correct